mods/butterflies/init.lua

```diff
@@ -47,13 +47,16 @@ for i in ipairs (butter_list) do
 			local player_name = placer and placer:get_player_name() or ""
 			local pos = pointed_thing.above
 
-			if not minetest.is_protected(pos, player_name) and
-					not minetest.is_protected(pointed_thing.under, player_name) and
-					minetest.get_node(pos).name == "air" then
-				minetest.set_node(pos, {name = "butterflies:butterfly_"..name})
+			if minetest.is_protected(pos, player_name) or minetest.is_protected(pointed_thing.under, player_name) then
```

Comment:
This protection check might not be necessary at all. `core.item_place_node` (called by `minetest.item_place`) already performs protection checks.

Reply:
I wasn't sure if the check was done so I kept